File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    
    min_rank = 0
    win_num = {6: 1, 5: 2, 4: 3, 3: 4, 2: 5, 1: 6, 0: 6}
    for lotto in lottos:
        if lotto in win_nums:
            min_rank += 1
    
    
    return [win_num[min_rank + lottos.count(0)], win_num[min_rank]]

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
import java.util.*;
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        
        int cnt = 0;
        int zero = 0;
        for(int lotto: lottos) {
            if (lotto == 0) {
                zero++;
                continue;
            }
            
            for(int win: win_nums) {
                if (lotto == win) {
                    cnt++;
                    break;
                }
            }
        }
        
        // System.out.println(cnt + "," + zero);
        // System.out.println(7 - (cnt + zero)  + ", " + (7 - cnt));
        answer[0] = Math.min(7 - (cnt + zero), 6);
        answer[1] = Math.min(7 - cnt, 6);

        return answer;
    }
}